nixos/modules/services/web-apps/matomo.nix

```diff
@@ -113,13 +113,22 @@ in {
           }
         '';
         description = ''
-            With this option, you can customize an nginx virtualHost which already has sensible defaults for Matomo.
-            Either this option or the webServerUser option is mandatory.
-            Set this to {} to just enable the virtualHost if you don't need any customization.
-            If enabled, then by default, the <option>serverName</option> is
-            <literal>''${user}.''${config.networking.hostName}.''${config.networking.domain}</literal>,
-            SSL is active, and certificates are acquired via ACME.
-            If this is set to null (the default), no nginx virtualHost will be configured.
+          With this option, you can customize an nginx virtualHost which already has sensible defaults for Matomo.
+          Either this option or the webServerUser option is mandatory.
+          Set this to {} to just enable the virtualHost if you don't need any customization.
+          If enabled, then by default, the <option>serverName</option> is
+          <literal>''${user}.''${config.networking.hostName}.''${config.networking.domain}</literal>,
+          SSL is active, and certificates are acquired via ACME.
+          If this is set to null (the default), no nginx virtualHost will be configured.
+        '';
+      };
+
+      mariaDB = mkOption {
```

Comment:
Often times we wrap this behind a few options like `database.createLocally`, `database.name`, etc...

... if you're interested.

Reply:
Making this an attrset from the beginning is a good idea for backcomp! Did that, added `name` option. Sadly there's no good way to configure this declaratively for Matomo (we may implement this in the future, but it's difficult), but one can at least change created local database name.